osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SecretManagementService.java

```diff
@@ -1,181 +1,317 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 @Slf4j
 public class SecretManagementService implements SecurityKeyService {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(SecretManagementService.class);
-    private final RsaEncryptionProvider rsaEncryptionProvider;
+    private final RsaEncrypter soapRsaEncrypter;
     private final SecretManagementClient secretManagementClient;
 
-    public SecretManagementService(RsaEncryptionProvider rsaEncryptionProvider,
-            SecretManagementClient secretManagementClient) {
-        this.rsaEncryptionProvider = rsaEncryptionProvider;
+    @Autowired
+    private RsaEncryptionService rsaEncryptionService;
+
+    @Autowired
+    private EncryptionService aesEncryptionService;
+
+    public SecretManagementService(RsaEncrypter soapRsaEncrypter, SecretManagementClient secretManagementClient) {
+        this.soapRsaEncrypter = soapRsaEncrypter;
         this.secretManagementClient = secretManagementClient;
     }
 
+    /**
+     * Re-encrypts the given key with a secret known only inside this protocol
+     * adapter.
+     * <p>
+     * New keys can be provided to OSGP from outside in a form encrypted with
+     * the public key from an asymmetrical key pair for the platform, which is
+     * available to external organizations.<br>
+     * Inside the DLMS protocol adapter keys are encrypted with a faster
+     * symmetrical encryption using a secret key that is not supposed to be
+     * known outside this protocol adapter.
+     *
+     * @param externallyEncryptedKey
+     *         key encrypted with the externally known public key for OSGP
+     *
+     * @return the key encrypted with the symmetrical secret key used only
+     *         inside the DLMS protocol adapter, or an empty byte array if
+     *         {@code externallyEncryptedKey == null}
+     *
+     * @throws FunctionalException
+     *         in case of a encryption/decryption errors while handling the
+     *         key
+     */
     @Override
-    public byte[] reEncryptKey(byte[] externallyEncryptedKey, SecurityKeyType keyType) throws FunctionalException {
-        throw new NotImplementedException();
-    }
+    public byte[] reEncryptKey(final byte[] externallyEncryptedKey/*, final SecurityKeyType keyType*/)
+            throws FunctionalException {
 
-    @Override
-    public byte[] decryptKey(byte[] encryptedKey, SecurityKeyType keyType) throws ProtocolAdapterException {
-        throw new NotImplementedException();
-    }
+        if (externallyEncryptedKey == null) {
+            return new byte[0];
+        }
 
-    @Override
-    public byte[] encryptKey(byte[] plainKey, SecurityKeyType keyType) throws ProtocolAdapterException {
-        throw new NotImplementedException();
-    }
+        final byte[] key = this.rsaDecrypt(externallyEncryptedKey);//, keyType);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)